### PR TITLE
Add doc covering when module parameters are set

### PR DESCRIPTION
**Pull request by adamrk**
Add documentation with guidance on allocating when setting module parameters and change the StringParam implementation to handle allocation failures.

### CODE REVIEW

**Comment by adamrk** [comment]
Check failures seem to be due to the llvm mirror sync in progress.

**Comment by ojeda** [comment]
Yeah, please feel free to force-push after a while.

File: rust/kernel/module_param.rs [hunk 1]
```diff
@@ -455,13 +467,14 @@ impl ModuleParam for StringParam {
     fn try_from_param_arg(arg: Option<&'static [u8]>) -> Option<Self> {
         // SAFETY: It is always safe to call [`slab_is_available`](../../../include/linux/slab.h).
         let slab_available = unsafe { crate::bindings::slab_is_available() };
-        arg.map(|arg| {
+        arg.and_then(|arg| {
             if slab_available {
                 let mut vec = alloc::vec::Vec::new();
+                vec.try_reserve(arg.len()).ok()?;
```
Comment (ojeda) [inline rust/kernel/module_param.rs]: Should this be `try_reserve_exact()`? i.e. do we expect further allocations?

**Review by ojeda — APPROVED** [review]

File: rust/kernel/module_param.rs (hunk 1 above)
Comment (adamrk) [reply rust/kernel/module_param.rs]: Ah, yep. `exact` is better.

**Comment by ojeda** [comment]
LGTM

**Pull request merged**